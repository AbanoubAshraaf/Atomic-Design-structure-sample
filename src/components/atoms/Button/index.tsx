import React from "react";

interface ButtonProps {
  text: string;
  onClick: () => void;
}

const button: React.FC<ButtonProps> = ({ text, onClick }) => {

  return (
    <button className="btn btn-primary" onClick={onClick}>
      {text}
    </button>
  );
};

export default button;
