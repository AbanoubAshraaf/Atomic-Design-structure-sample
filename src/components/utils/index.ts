export const isValidAge = (age:number): boolean => {
    return (
      age > 0 && // Age must be positive
      !isNaN(age) && // Age must be a valid number
      Number.isInteger(age) // Age must be an integer
    );
  };
  