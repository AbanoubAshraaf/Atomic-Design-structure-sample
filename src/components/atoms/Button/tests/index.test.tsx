import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "..";
import { createSnapShot } from "@/components/test/utils";

describe("<Button />", () => {
  test("render button with text", () => {
    const { container, getByText } = render(<Button text="Click Me" onClick={() => { }} />);
    expect(getByText("Click Me")).toBeInTheDocument();
    createSnapShot(container)
  });

  test("call onClick handler", () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Button text="Click Me" onClick={handleClick} />);
    fireEvent.click(getByText("Click Me"));
    expect(handleClick).toHaveBeenCalled();
  });
});
