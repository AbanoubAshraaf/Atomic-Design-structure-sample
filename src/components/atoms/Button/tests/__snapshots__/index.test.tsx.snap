// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render button with text 1`] = `
<div>
  <button
    class="btn btn-primary"
  >
    Click Me
  </button>
</div>
`;
